version: '3.4'

services:
  rabbitmq:
    container_name: rabbitmq
    image: rabbitmq:management-alpine
    ports:
      - "15672:15672"
      - "5672:5672"
    healthcheck:
        test: rabbitmq-diagnostics -q ping
        interval: 30s
        timeout: 30s
        retries: 3
    networks: 
      - admin_net-network
    
  postgres-db:
    container_name: postgres-db
    image: postgres:14.1
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_PASSWORD= "myPassw0rd"
    volumes:
      - postgres_data:/var/opt/postgres
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5  
    networks: 
      - admin_net-network
  
  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
      - PGADMIN_DEFAULT_EMAIL= ${PGADMIN_DEFAULT_EMAIL:-pgadmin4@pgadmin.org}
      - PGADMIN_DEFAULT_PASSWORD= ${PGADMIN_DEFAULT_PASSWORD:-admin}
    ports:
      - "${PGADMIN_PORT:-5050}:80"
    volumes:
       - pgadmin_data:/var/lib/pgadmin
    networks: 
      - admin_net-network
      
  seq:
    image: datalust/seq
    container_name: seq
    restart: always
    environment:
      - ACCEPT_EULA=Y
    ports:
      - 5380:80
      - 5341:5341
    volumes:
      - ./.seqlogs:/data
    networks: 
      - admin_net-network
      
volumes:
  postgres_data:
  pgadmin_data:

networks:
  admin_net-network:
    external: true